code/obj/machinery/door/door_control.dm

```diff
@@ -960,15 +954,8 @@
 					return ..()
 				if (!access_type)
 					open_door()
-				else if (!access_type_secondary)
-					open_door()
-				else if (!access_type_secondary)
-					open_door()
 				else
-					if(V.com_system.access_type == src.access_type)
-						open_door()
-
-					else if(V.com_system.access_type_secondary == src.access_type_secondary)
+					if(V.com_system.access_type.Find(src.access_type))
```

Comment:
does .Find work on non-lists like this?

Reply:
It's not intended to be a non-list, but if it isn't this would cause a runtime, yeah.
I could add functionality for it being a single value, I guess.

Reply:
i see - i thought you had defined it as a non-list but that was the var also named `access_type` on the door controllers.